design/general_refactorings.md

```diff
@@ -0,0 +1,49 @@
+Before refactor concrete thing here is a list of things that need to be done all over the code:
+
+1. Go Over All Namings And Improve them
+
+2. Directory Organize:
+
+   Current Directories
+
+   | Directory Name | Porpuse                           |
+   | :------------- | :-------------------------------- |
+   | host           | files related to host command     |
+   | init           | files related to init command     |
+   | list           | files related to list command     |
+   | login          | files related to login command    |
+   | logout         | files related to logout command   |
+   | policy         | files related to policy command   |
+   | user           | files related to user command     |
+   | variable       | files related to variable command |
+
+   
+
+   And this files are without a folder : api.py, client.py, argparse_wrapper.py, cli.py , client.py, config.py, constants.py, credentials_data.py, credentials_form_file.py, endpoint.py, errors.py, http_wrapper.py, ssl_service.py, version.py
+
+   The change i want to make is to organize the files by their types and not by the command they do and have directories that can be used for the files without directory:
```

Comment:
Recommendation:

Organize the files by their types and not by their command. For example, create Controller, ... folders that will combine all Controllers together regardless of the command they serve

Reply:
I would organize the doc according to the following 

_Proposal_
Create...

_Recommendation_
Update..

So that each point will be contained in its own subsection